backend/Dockerfile

```diff
@@ -0,0 +1,41 @@
+# build
+FROM golang:1.23.4-alpine AS builder
+
+WORKDIR /app
+
+COPY go.mod go.sum ./
+
+RUN go mod download
+
+COPY . .
+
+ARG COMMIT_SHA=unknown
+ARG BUILD_TIME=unknown
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
+    -ldflags "-w -s -X 'backend/internal/version.commitSHA=${COMMIT_SHA}' -X 'backend/internal/version.buildTime=${BUILD_TIME}'" \
+    -o bin/backend \
+    ./cmd/server
+
+# run
+FROM alpine:3.19

```

Comment:
Maybe something to look at later - Alpine is generally not recommended these days. While it can keep built image size small, it replaces `glibc` with `musl`; which can cause issues at runtime.

Common recommendation is to either use an Ubuntu LTS or Google's distroless.